include/deal.II/physics/elasticity/kinematics.h

```diff
@@ -0,0 +1,278 @@
+
+      /**
+       * Returns the small strain SymmetricTensor,
+       * as constructed from the displacement gradient tensor @p Grad_u.
+       * The result is expressed as
+       * @f[
+       *  \mathbf{e} := \frac{1}{2} \left[ \nabla_{0}\mathbf{u}
+       *   + [\nabla_{0}\mathbf{u}]^{T} \right] \, .
+       * @f]
+       * where $\mathbf{u} = \mathbf{u}(\mathbf{X})$ is the displacement at position
+       * $\mathbf{X}$ in the referential configuration.
+       * The differential operator $\nabla_{0}$ is defined as
+       * $\frac{\partial}{\partial \mathbf{X}}$.
+       */
+      static SymmetricTensor<2, dim, Number>
+      e (const Tensor<2, dim, Number> &Grad_u);
+
+      /**
+       * Returns the Almansi strain SymmetricTensor,
+       * as constructed from the deformation gradient tensor @p F.
+       * The result is expressed as
+       * @f[
+       *  \mathbf{A} := \frac{1}{2} \left[ \mathbf{I}
+       *   - \mathbf{F}^{-T}\cdot\mathbf{F}^{-1} \right] \, .
+       * @f]
+       */
+      static SymmetricTensor<2, dim, Number>
+      a (const Tensor<2, dim, Number> &F);
+
+
+     /**
+      * Returns the spatial velocity gradient Tensor,
+      * as constructed from the deformation gradient tensor @p F
+      * and its rate @p dF_dt.
+      * The result is expressed as
+      * @f[
+      *  \mathbf{l} := \dot{\mathbf{F}}\cdot\mathbf{F}^{-1} \, .
+      * @f]
+      */
+     static Tensor<2, dim, Number>
+     l (const Tensor<2,spacedim> & F,
+        const Tensor<2,spacedim> & dF_dt);
+
+     /**
+      * Returns the rate of deformation Tensor (also known as the rate of strain
+      * Tensor), as constructed from the deformation gradient tensor @p F
+      * and its rate @p dF_dt.
+      * The result is expressed as
+      * @f[
+      *  \mathbf{d} := \frac{1}{2} \left[ \mathbf{l} + \mathbf{l}^{T} \right] \, .
+      * @f]
+      * where $\mathbf{l}$ is the spatial velocity gradient Tensor.
+      */
+     static SymmetricTensor<2, dim, Number>
+     d (const Tensor<2,spacedim> & F,
+        const Tensor<2,spacedim> & dF_dt);
+
+     /**
+      * Returns the rate of rotation Tensor (also known as the vorticity
+      * Tensor), as constructed from the deformation gradient tensor @p F
+      * and its rate @p dF_dt.
+      * The result is expressed as
+      * @f[
+      *  \mathbf{w} := \frac{1}{2} \left[ \mathbf{l} - \mathbf{l}^{T} \right] \, .
+      * @f]
+      * where $\mathbf{l}$ is the spatial velocity gradient Tensor.
+      */
+     static Tensor<2, dim, Number>
+     w (const Tensor<2,spacedim> & F,
+        const Tensor<2,spacedim> & dF_dt);
+    };
+
+  }
+}
+
+
+
+// ------------------------- inline functions ------------------------
+
+
+
+template <int dim, typename Number>
+inline
+Tensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::F (const Tensor<2, dim, Number> &Grad_u)
+{
+// See Wriggers (2008) p 23 equ. 3.14
+  return Physics::Elasticity::StandardTensors<dim>::I + Grad_u;
```

Comment:
I vaguely remember this biting me in the past so I usually do the equivalent of `static_cast<Tensor<...>>(I) + Grad_u`. It may have been a red-herring from me trying to fix something else, but it's worth a double check.